lib/OpenQA/Worker/Jobs.pm

```diff
@@ -126,14 +126,15 @@ sub stop_job {
     my ($aborted, $job_id) = @_;
 
     # we call this function in all situations, so better check
-    if (!$job) {
-        # in case there is no job, we can stop ther worker
-        # asap, otherwise wait for the actual stop to finish before
-        # do the IOloop stop
+    if (!$job || $stop_job_running) {
+        # in case there is no job, or if the job was asked to stop
+	# we stop the worker asap, otherwise wait for the actual 
+	# stop to finish before, or run into a condition where the job
+	# runs forever and worker Mojo::IOLoop->stop is never called
+        log_debug("Either there is no job running or we were asked to stop");
```

Comment:
So it will be shown like this:

 Either there is no job running (1) or we were asked to stop (no tests scheduled) There's no reason to show $job as $stop_job_running will be undef or 0 in the other case :)